### DISCUSSION TITLE
About File Handling

### DISCUSSION THREAD

**Discussion by GabrielRybak** [shish/shimmie2#1126] [Q&A]
Hello,

I’m a junior sysadmin, and I’ve been learning a lot from your project. Thank you for all your hard work! I’m currently hosting a joke booru for myself and some friends.

I have a couple of questions related to file management:

Storage Location: After users upload media files (such as pictures, mp4, and webms) via the web interface, are these files stored locally on the disk or in the database?

Loading Existing Content: Is there a way to load existing content directly from my disk instead of uploading it through the web interface? For instance, I have over 500 pictures in /home/myuser/pictures. Uploading them individually seems redundant, and I’d prefer to keep them on the disk rather than in the database.

Any insights you can provide would be greatly appreciated!

**Comment by GabrielRybak** [comment]
Update: I installed phpmyadmin and I see that media content is stored inside the shimmie db so that answers my first question!
I think this discards the possibility of adding direct content from my disk in a simple or direct way, they all should be loaded to the database and then create the corresponding entrances for tags and other metadata. Can you confirm this? Apologies for my novice question!

This leaves me with two options, duplicating the used diskspace for the files and the database, or loading everything into the database and deleting the original files from disk. Am I correct?

Thank you for any tip!

**Comment by shish** [comment]
> Storage Location: After users upload media files (such as pictures, mp4, and webms) via the web interface, are these files stored locally on the disk or in the database?

The files are stored in `<whichever folder shimmie is installed>/data/images`, with metadata (width, height, filename, hash, upload date, etc) stored in the database for efficient searching


> Loading Existing Content: Is there a way to load existing content directly from my disk instead of uploading it through the web interface?

There is a "bulk add" extension which you can point at a folder on the server's disk, and it will import the files from there into itself


However a file is imported into shimmie, the site will keep its own copy of the files - being able to serve any pre-existing file from anywhere on the server's disk would be technically possible (as in, if you rewrote the file-serving code to do so), but comes with a bunch of practical and security issues
